File: digitConverter.py
#!/bin/python
# from math import pow

# Binary


""" def fromBinary(binary):
    Binary = []
    for i in range(0, len(str(binary))):
        Binary.append(int(str(binary)[i]))
    Binary.reverse()
    dec = 0
    for i in range(0, len(str(binary))):
        dec = dec + (int(Binary[i]) * pow(2, i))
    return int(dec) """


def fromBinary(binary: int) -> int:
    """ From binary takes an int binary strand and converts it into a decimal number that represents the strand """
    bin = int(binary)
    dec = exp = 0
    while (bin > 0):
        dec = dec + int((bin % 10) * 2 ** exp)
        bin //= 10
        exp += 1
    return dec


""" def toBinary(dec):
    dec = int(dec)
    Binary = []
    while (dec > 0):
        Binary.append(str(dec % 2))
        dec = ((dec - (dec % 2)) / 2)
    Binary.reverse()
    binary = ""
    for i in range(0, len(Binary)):
        binary = (str(binary) + str(Binary[i]))
    return int(binary) """

def toBinary(dec: int) -> int:
    """ To Binary take a decimal number and converts it into an int Binary strand """
    dec = int(dec)
    binary = ""
    while (dec > 0):
        binary = str(dec % 2) + binary
        dec //= 2
    return int(binary)
# /Binary
# Octal


""" def fromOctal(octal):
    Octal = []
    for i in range(0, len(str(octal))):
        Octal.append(int(str(octal)[i]))
    Octal.reverse()
    dec = 0
    for i in range(0, len(str(octal))):
        dec = dec + (int(Octal[i]) * pow(8, i))
    return int(dec) """


def fromOctal(octal: int) -> int:
    """ From Octal takes a strand in octal and converts it into a decimal number """
    oct = int(octal)
    dec = exp = 0
    while (oct > 0):
        dec = dec+int((oct % 10) * 8 ** exp)
        # print("oct:", oct)
        # print("dec:",dec)
        oct //= 10
        exp += 1
        # print("exp:",exp)
    return int(dec)


""" def toOctal(dec):
    dec = int(dec)
    Octal = []
    while (dec > 0):
        Octal.append(str(dec % 8))
        dec = ((dec - (dec % 8)) / 8)
    Octal.reverse()
    octal = ""
    for i in range(0, len(Octal)):
        octal = (str(octal) + str(Octal[i]))
    return int(octal) """


def toOctal(dec: int) -> int:
    """ To Octal takes a decimal number and converts it into an octal strand """
    dec = int(dec)
    octal = ""
    while (dec > 0):
        octal = str(dec % 8) + octal
        dec //= 8
    return int(octal)
# /Octal
# Decimal

# /Decimal
# Hexadecimal

# /Hexadecimal

File: DigitConverter.java
import java.util.ArrayList;
public class DigitConverter {
	private static int power(int base, int exp){
		int result = 1;
		for(int i=0; i<exp; i++)
			result *= base;
		return result;
	}
	public static int toBin(int decimal){
		//char[] storeBin = {};
		ArrayList<Integer> storeBin = new ArrayList<Integer>();
		String result = "";
		while (decimal > 0){
			int remain = decimal%2;
			storeBin.add(remain);
			decimal = (decimal - remain)/2;
		}
		for (int i = storeBin.size()-1; i >= 0; i--)
			result = result + storeBin.get(i).toString();
		return Integer.valueOf(result);
	}
	public static int fromBin(int binary){
		String bin = String.valueOf(binary);
		int[] storeBin = new int[bin.length()];
		int result = 0;
		for (int i = 0; i < bin.length(); i++)
			storeBin[i] = Integer.valueOf(bin.substring(bin.length()-1-i,bin.length()-i));
		for (int i = 0; i < storeBin.length; i++)
			result += storeBin[i] * power(2,i);
		return result;
	}
	public static int toOct(int decimal){
		//char[] storeBin = {};
		ArrayList<Integer> storeOct = new ArrayList<Integer>();
		String result = "";
		while (decimal > 0){
			int remain = decimal%8;
			storeOct.add(remain);
			decimal = (decimal - remain)/8;
		}
		for (int i = storeOct.size()-1; i >= 0; i--)
			result = result + storeOct.get(i).toString();
		return Integer.valueOf(result);
	}
	public static int fromOct(int octal){
		String oct = String.valueOf(octal);
		int[] storeOct = new int[oct.length()];
		int result = 0;
		for (int i = 0; i < oct.length(); i++)
			storeOct[i] = Integer.valueOf(oct.substring(oct.length()-1-i,oct.length()-i));
		for (int i = 0; i < storeOct.length; i++)
			result += storeOct[i] * power(2,i);
		return result;
	}
	public static String toHex(int decimal){
		//char[] storeBin = {};
		ArrayList<Integer> storeHex = new ArrayList<Integer>();
		String result = "";
		while (decimal > 0){
			int remain = decimal%16;
			storeHex.add(remain);
			decimal = (decimal - remain)/16;
		}
		for (int i = storeHex.size()-1; i >= 0; i--){
			switch (storeHex.get(i)) {
				case 10:
					result += "A";
					break;
				case 11:
					result += "B";
					break;
				case 12:
					result += "C";
					break;
				case 13:
					result += "D";
					break;
				case 14:
					result += "E";
					break;
				case 15:
					result += "F";
					break;
				default:
					result += storeHex.get(i).toString();
					break;
			}
		}
		return result;
	}
	public static int fromHex(String hex){
		int[] storeHex = new int[hex.length()];
		int result = 0;
		for (int i = 0; i < hex.length(); i++){
			switch (hex.substring(hex.length()-1-i,hex.length()-i).toLowerCase()) {
				case "a":
					storeHex[i] = 10;
					break;
				case "b":
					storeHex[i] = 11;
					break;
				case "c":
					storeHex[i] = 12;
					break;
				case "d":
					storeHex[i] = 13;
					break;
				case "e":
					storeHex[i] = 14;
					break;
				case "f":
					storeHex[i] = 15;
					break;
				default:
					storeHex[i] = Integer.valueOf(hex.substring(hex.length()-1-i,hex.length()-i));
					break;
			}
		}
		for (int i = 0; i < storeHex.length; i++)
			result += storeHex[i] * power(16,i);
		return result;
	}
	public static void main(String[] args){
		if (args[0].equals("tobin")){
			System.out.println("ToBinary: "+args[1]);
			System.out.println(toBin(Integer.valueOf(args[1])));
		}
		else if (args[0].equals("frombin")){
			System.out.println("FromBinary: "+args[1]);
			System.out.println(fromBin(Integer.valueOf(args[1])));
		}
		else if (args[0].equals("tooct")){
			System.out.println(toOct(Integer.valueOf(args[1])));
		}
		else if (args[0].equals("fromoct")){
			System.out.println(fromOct(Integer.valueOf(args[1])));
		}
		else if (args[0].equals("tohex")){
			System.out.println(toHex(Integer.valueOf(args[1])));
		}
		else if (args[0].equals("fromhex")){
			System.out.println(fromHex(args[1]));
		}
	}
}
